### PR TITLE
#34: Rename wrongly named packages

### LINKED ISSUE #34: Rename wrongly named packages
## Background
Some packages use non-standard names, containing underscores.

## Feature
Rename the packages to follow the naming conventions.
Considering renaming the `DBFunction` successor classes.

## Example [Optional]
`naming_conventions`
`letters_case`

### PR DESCRIPTION
**Pull request by benedeki**
* renamed package `naming_conventions` to `naming` and reorganized the files within
* renamed package `statushandling` to `status` and reorganized the files within
* renamed `DBSeqFunction` to `DBMultipleResultFunction`
* renamed `DBUniqueFunction` to `DBSingleResultFunction`
* renamed `DBOptionFunction` to `DBOptionalResultFunction`
* renamed the execution methods within `DBFunction` to match the above successor names

Closes #34

### CODE REVIEW

File: slick/src/main/scala/za/co/absa/fadb/slick/SlickPgFunctionWithStatusSupport.scala [hunk 1]
```diff
@@ -17,21 +17,24 @@
 package za.co.absa.fadb.slick
 
 import slick.jdbc.{GetResult, PositionedResult}
-import za.co.absa.fadb.statushandling.FunctionStatus
+import za.co.absa.fadb.status.FunctionStatus
+import za.co.absa.fadb.status.handling.StatusHandling
```
Comment (miroslavpojer) [inline slick/src/main/scala/za/co/absa/fadb/slick/SlickPgFunctionWithStatusSupport.scala:21]: Not used import.

**Review by miroslavpojer — COMMENTED** [review]
- [ ] code pulled
- [ ] built ==> tests fails after merge
- [ ] code review
- [ ] test ran

File: slick/src/main/scala/za/co/absa/fadb/slick/SlickPgFunctionWithStatusSupport.scala (hunk 1 above)
Comment (benedeki) [inline slick/src/main/scala/za/co/absa/fadb/slick/SlickPgFunctionWithStatusSupport.scala:21]:
```suggestion
import za.co.absa.fadb.status.FunctionStatus
```

File: core/src/test/scala/za/co/absa/fadb/naming_conventions/SnakeCaseNamingSuite.scala [hunk 2]
```diff
@@ -2,7 +2,8 @@ package za.co.absa.fadb.naming_conventions
 
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
-import za.co.absa.fadb.naming_conventions.letters_case.LettersCase._
+import za.co.absa.fadb.naming.implementations.SnakeCaseNaming
```
Comment (lsulak) [inline core/src/test/scala/za/co/absa/fadb/naming_conventions/SnakeCaseNamingSuite.scala:5]: The package is still in `za.co.absa.fadb.naming_conventions`

File: core/src/test/scala/za/co/absa/fadb/DBSchemaSuite.scala [hunk 3]
```diff
@@ -16,7 +16,7 @@
 package za.co.absa.fadb
 
 import org.scalatest.funsuite.AnyFunSuite
-import za.co.absa.fadb.naming_conventions.SnakeCaseNaming.Implicits.namingConvention
+import za.co.absa.fadb.naming.implementations.SnakeCaseNaming.Implicits.namingConvention
 
 import scala.concurrent.{Await, ExecutionContext, Future}
```
Comment (lsulak) [inline core/src/test/scala/za/co/absa/fadb/DBSchemaSuite.scala:21]: here the Await is not used

**Review by lsulak — COMMENTED** [review]
The review finished, I really really like the new naming.
- [x]  code reviewed
- [x]  pulled
- [x]  built
- [x]  ran tests

I found a few minor issues - once you address them, I'm happy to approve it!

File: core/src/test/scala/za/co/absa/fadb/naming_conventions/ExplicitNamingRequiredSuite.scala [hunk 4]
```diff
@@ -3,6 +3,7 @@ package za.co.absa.fadb.naming_conventions
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
 import za.co.absa.fadb.exceptions.NamingException
+import za.co.absa.fadb.naming.ExplicitNamingRequired
```
[the remark this answers is not in the document]
Comment (benedeki) [reply core/src/test/scala/za/co/absa/fadb/naming_conventions/ExplicitNamingRequiredSuite.scala]: Ha, refactor, didn't work on tests 😮

**Review by lsulak — APPROVED** [review]

**Pull request merged**